zebra-chain/src/primitives/address.rs

```diff
@@ -119,4 +119,28 @@ impl Address {
     pub fn is_transparent(&self) -> bool {
         matches!(self, Self::Transparent(_))
     }
+
+    /// Returns the payment address for transparent or sapling addresses.
+    #[allow(clippy::unwrap_in_result)]
+    pub fn payment_address(&self) -> Option<String> {
+        use zcash_address::{ToAddress, ZcashAddress};
+
+        match &self {
+            Self::Transparent(address) => Some(address.to_string()),
+            Self::Sapling { address, network } => {
+                let data = address.to_bytes();
+                let address =
+                    ZcashAddress::from_sapling(Self::zcash_address_network(*network), data);
+                Some(address.encode())
+            }
+            Self::Unified { .. } => None,
```

Comment:
Optional: After PR #6185 merges, the correct implementation for the most preferred unified address is:

```suggestion
            Self::Unified { network, address } => Some(address.first().encode(network)),
```

This lets us drop the `Option` from the return value. But we can do this in a follow-up PR.